build/jobs/add-aad-test-environment.yml

```diff
@@ -36,34 +36,21 @@ steps:
           resource   = $resource
       }
 
-      # If a deleted keyvault exists, remove it first
+      # If a deleted keyvault with purge protection exists, try to restore it.
       $environmentName = "$(DeploymentEnvironmentName)".ToLower() -replace "\.", "" 
       Write-Host "Installed module and set variables"
 
       $vaultName = "${environmentName}-ts"
-      if (Get-AzKeyVault -VaultName $vaultName -Location "westus" -InRemovedState)
+      $vaultLocation = "westus"
```

Comment:
NIT - would be nice to have the region parameterized while you are here. 😄 Not required as it wasn't parameterized before.